classes/blocks/class-base-block.php

```diff
@@ -94,4 +96,74 @@ protected function is_rest_request() {
 	public static function update_data( array $fields ) {
 		throw new NotImplemented( 'Method update_data is not implemented for ' . static::class );
 	}
+
+	/**
+	 * Register scripts and styles for a block
+	 * 
+	 * @param array $editor_script_dependencies Dependencies for the Editor script.
+	 * @param array $script_dependencies Dependencies for the Frontend script.
+	 */
+	public static function enqueue_editor_assets() {
+		$camelized_block_name = str_replace("-", "", ucwords(static::BLOCK_NAME, "-"));
+
+		// Editor Script
+		$file_url = trailingslashit( P4GBKS_PLUGIN_URL ) . 'assets/build/' . $camelized_block_name . 'EditorScript.js';
+		$file_path = trailingslashit( P4GBKS_PLUGIN_DIR ) . 'assets/build/' . $camelized_block_name . 'EditorScript.js';
+		wp_enqueue_script(
+			static::BLOCK_NAMESPACE_PREFIX . '/' . static::BLOCK_NAME . '-editor-script',
+			$file_url,
+			'planet4-blocks-editor-script',
+			\P4GBKS\Loader::file_ver($file_path),
+			true
+		);
+		
+		// Editor Style
+		$file_url = trailingslashit( P4GBKS_PLUGIN_URL ) . 'assets/build/' . $camelized_block_name . 'EditorStyle.min.css';
+		$file_path = trailingslashit( P4GBKS_PLUGIN_DIR ) . 'assets/build/' . $camelized_block_name . 'EditorStyle.min.css';
+		wp_enqueue_style(
+			static::BLOCK_NAMESPACE_PREFIX . '/' . static::BLOCK_NAME . '-editor-style',
+			$file_url,
+			[],
+			\P4GBKS\Loader::file_ver($file_path),
+		);
+			
+		// Frontend Style (aka: style)
+		$file_url = trailingslashit( P4GBKS_PLUGIN_URL ) . 'assets/build/' . $camelized_block_name . 'Style.min.css';
+		$file_path = trailingslashit( P4GBKS_PLUGIN_DIR ) . 'assets/build/' . $camelized_block_name . 'Style.min.css';
+		wp_enqueue_style(
+			static::BLOCK_NAMESPACE_PREFIX . '/' . static::BLOCK_NAME . '-style',
+			$file_url,
+			[],
+			\P4GBKS\Loader::file_ver($file_path),
+		);
+	}
+
+	public static function enqueue_frontend_assets() {
+		if ( has_block( static::BLOCK_NAMESPACE_PREFIX . '/' . static::BLOCK_NAME ) ) {
```

Comment:
We should keep an eye on the performance of this, since it does parse the post content again for each type of block we have, so with more types of blocks and lengthy post content this might not scale well. Maybe we can set it up in a way so that we parse blocks once and then search the result for each block type?

Could be `parse_blocks` performance is good enough and this is a total non issue.

Reply:
Ah nvm it internally just does a regex, should be fine :+1: